software/messaging/src/main/java/brooklyn/entity/messaging/kafka/KafkaBrokerImpl.java

```diff
@@ -99,35 +100,43 @@ protected void connectSensors() {
-                        .onException(Functions.constant(-1l)))
+                        .onException(Functions.constant(-1l))
+                        .enabled(retrieveUsageMetrics))
                 .pollAttribute(new JmxAttributePollConfig<Long>(TOTAL_FETCH_TIME)
                         .objectName(SOCKET_SERVER_STATS_MBEAN)
                         .attributeName("TotalFetchRequestMs")
-                        .onException(Functions.constant(-1l)))
+                        .onException(Functions.constant(-1l))
+                        .enabled(retrieveUsageMetrics))
                 .pollAttribute(new JmxAttributePollConfig<Double>(MAX_FETCH_TIME)
                         .objectName(SOCKET_SERVER_STATS_MBEAN)
                         .attributeName("MaxFetchRequestMs")
-                        .onException(Functions.constant(-1.0d)))
+                        .onException(Functions.constant(-1.0d))
+                        .enabled(retrieveUsageMetrics))
                 .pollAttribute(new JmxAttributePollConfig<Long>(PRODUCE_REQUEST_COUNT)
                         .objectName(SOCKET_SERVER_STATS_MBEAN)
                         .attributeName("NumProduceRequests")
-                        .onException(Functions.constant(-1l)))
+                        .onException(Functions.constant(-1l))
+                        .enabled(retrieveUsageMetrics))
                 .pollAttribute(new JmxAttributePollConfig<Long>(TOTAL_PRODUCE_TIME)
                         .objectName(SOCKET_SERVER_STATS_MBEAN)
                         .attributeName("TotalProduceRequestMs")
-                        .onException(Functions.constant(-1l)))
+                        .onException(Functions.constant(-1l))
+                        .enabled(retrieveUsageMetrics))
                 .pollAttribute(new JmxAttributePollConfig<Double>(MAX_PRODUCE_TIME)
                         .objectName(SOCKET_SERVER_STATS_MBEAN)
                         .attributeName("MaxProduceRequestMs")
-                        .onException(Functions.constant(-1.0d)))
+                        .onException(Functions.constant(-1.0d))
+                        .enabled(retrieveUsageMetrics))
                 .pollAttribute(new JmxAttributePollConfig<Long>(BYTES_RECEIVED)
                         .objectName(SOCKET_SERVER_STATS_MBEAN)
                         .attributeName("TotalBytesRead")
-                        .onException(Functions.constant(-1l)))
+                        .onException(Functions.constant(-1l))
+                        .enabled(retrieveUsageMetrics))
                 .pollAttribute(new JmxAttributePollConfig<Long>(BYTES_SENT)
                         .objectName(SOCKET_SERVER_STATS_MBEAN)
                         .attributeName("TotalBytesWritten")
-                        .onException(Functions.constant(-1l)))
+                        .onException(Functions.constant(-1l))
+                        .enabled(retrieveUsageMetrics))
```

Comment:
Should we have a global `enabled` option on the feed itself to cater for these cases where all `pollAttributes` are gated on the same condition?